ui/src/adapters/api/connections.ts

```diff
@@ -0,0 +1,87 @@
+import axios from "axios";
+import { z } from "zod";
+
+import { APIResponse, HTTPStatusSuccess, ResultOK, buildAPIError } from "src/adapters/api";
+import { Env } from "src/domain";
+import { API_VERSION, QUERY_SEARCH_PARAM } from "src/utils/constants";
+import { StrictSchema } from "src/utils/types";
+
+const buildAuthorizationHeader = (env: Env) =>
+  `Basic ${window.btoa(`${env.api.username}:${env.api.password}`)}`;
+
+//TODO move it to credentials when is properly cleaned
+export interface Credential {
+  attributes: {
+    type: string;
+  };
+  id: string;
+}
+
+export interface Connection {
+  createdAt: string;
+  credentials: Credential[];
+  id: string;
+  issuerID: string;
+  userID: string;
+}
+
+export async function getConnections({
+  credentials = false,
+  env,
+  params: { query },
+  signal,
+}: {
+  credentials?: boolean;
+  env: Env;
+  params: {
+    query?: string;
+  };
+  signal: AbortSignal;
+}): Promise<APIResponse<Connection[]>> {
+  try {
+    const response = await axios({
+      baseURL: env.api.url,
+      headers: {
+        Authorization: buildAuthorizationHeader(env),
+      },
+      method: "GET",
+      params: new URLSearchParams({
+        ...(query !== undefined ? { [QUERY_SEARCH_PARAM]: query } : {}),
+        ...(credentials ? { credentials: "true" } : {}),
+      }),
+      signal,
+      url: `${API_VERSION}/connections`,
```

Comment:
Any reason the version is here? All other actions have `baseURL: `${env.api.url}/${API_VERSION}`. We should probably provide a helper akin to `buildAPIURL(env)`.

Reply:
I'm following the documentation where the version is related to the path
![image](https://user-images.githubusercontent.com/19190248/228521933-db0c86b1-0123-48b6-838b-077d8abe84a9.png)

Reply:
@adrifdez I am referring to what Alex is pointing out, namely to put `API_VERSION` into `baseURL` instead of here in `url`, then to apply this globally via a helper, similar to what you did with `buildAuthorizationHeader()` (calling it e.g. `buildAPIURL()`).

Reply:
I've solved the problem that Alex mentioned, but I don't see the need to create a function and import it just to add the version.

Reply:
agreed

Reply:
I'm not saying what is best or more appropriate, just keeping align with the way that the documentation explain the endpoints

Reply:
Okay, fine then.